src/generator_files.rs

```diff
@@ -161,11 +161,10 @@ pub fn orchestrate(client: &DatagoClient) -> DatagoEngine {
     }));
 
     DatagoEngine {
-        pages_rx: samples_metadata_rx_worker,
+        samples_metadata_rx,
         samples_tx,
         samples_rx,
-        pinger: None,
-        feeder: generator,
+        feeder,
```

Comment:
just two levels everywhere now:
- samples_metadata (enough to materialize them) and samples (materialized)
- feeder and workers

Reply:
note that the worker thread is attached to a multithread tokio runtime, hence a worker pool for real